ccm_web/server/src/canvas/canvas.service.ts

```diff
@@ -241,4 +245,9 @@ export class CanvasService {
     const requestor = new CanvasRequestor(this.url + endpoint, token.accessToken, requestorOptions)
     return requestor
   }
+
+  async createRequestorForAdmin (endpoint: SupportedAPIEndpoint): Promise<CanvasRequestor> {
+    const requestor = new CanvasRequestor(this.url + endpoint, this.adminSecret, requestorOptions)
+    return requestor
+  }
```

Comment:
@ssciolla, I could use your advice here.  I think I'm misunderstanding how Canvas API authN works.

I thought getting an admin requestor would be as simple as passing the Canvas API key's secret to the requestor constructor.  However, whenever I use the requestor created by this in the call to `createExternalUser()` (see: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/247/files#diff-ef40e88ad430e58d8ff24581b5ad8d8084846e61af749e0159f63f740b0712ddR80), I always get an HTTP 401 error ("Unauthorized").

What is the correct way to use the Canvas API key I created for this?

Reply:
My misunderstanding here was that the API token needed to be provided here, not the secret for the developer key.  Normally the secret would be used in an OAuth process to obtain or refresh an API token.  In this case, where the admin user wouldn't be available to approve the OAuth request, that would require awkward configuration and setup steps.  It would also introduce a race condition when multiple near-simultaneous requests would trigger overlapping refresh requests.  The race condition could not be completely mitigated.  As a result, the code will use a static API token that doesn't require refreshing.